### PR TITLE
Improve external wmts management for 3d  

### LINKED ISSUE #3830: WMTS layers aren't displayed in 3D
https://s.geo.admin.ch/738d9da4c1

### LINKED ISSUE #3860: [print] Client-side reprojected WMS and WMTS won't print
WMS and WMTS layers from server not supporting EPSG:21781 are reprojected on-the-fly client-side (if EPSG:4326 is supported).
https://s.geo.admin.ch/73db2f5516

The MapFish Print Server v2 is not able to mix layer of different projection., so the layer won't print

```
java.io.IOException: Didn't receive an image while reading: 
http://www.webatlasde.de/arcgis/services/Maps4BW/MapServer/WMSServer?BBOX=595563.019945192%2C234802.71788842385%2C680441.3553455826%2C290894.3824880332&HEIGHT=1104&WIDTH=1671&map_resolution=150&CRS=EPSG%3A21781&MAP_RESOLUTION=150&SERVICE=WMS&REQUEST=GetMap&DPI=150&STYLES=&TRANSPARENT=true&EXCEPTIONS=XML&VERSION=1.1.1&FORMAT=image%2Fpng&format_options=dpi%3A150&LAYERS=Halbtonvariante&SRS=EPSG%3A21781
```

There is no obvious solution for this issue. We have to warn the user that the layer won't be printed.

### LINKED ISSUE #4030: Print: client-side reprojected external WMTS are not printed in PDF
Open
https://s.geo.admin.ch/76c339aad7

Print to PDF

Result: PDF does not contain external layer

### PR DESCRIPTION
**Pull request by oterral**
This PR:
   - find the good matrixset to use for 3d.
   - display an alert message if a layer can't be displayed in 3d (no matrixset found)
   - display an alert message if a layer can't be printed (layers using the on-the-fly reprojection)  
   - adds an html report available on dev only [Test](https://mf-geoadmin3.dev.bgdi.ch/ltteo/src/coverage/index.html)

[Test](https://mf-geoadmin3.int.bgdi.ch/teo_wmts3d/index.html)

Fix #4030 [Print the asitvd layer](https://mf-geoadmin3.int.bgdi.ch/teo_wmts3d/index.html?lang=fr&topic=ech&bgLayer=voidLayer&dev3d=true&zoom=7&debug&E=2614532.77&N=1178055.34&layers=WMTS%7C%7Casitvd.fond_pourortho%7C%7Chttp:%2F%2Fows.asitvd.ch%2Fwmts%3FSERVICE%3DWMTS%26REQUEST%3DGetCapabilities%26VERSION%3D1.0.0)
Fix #3860 [Display an alert message when a WMTS is not printable](https://mf-geoadmin3.int.bgdi.ch/teo_wmts3d/index.html?bgLayer=voidLayer&lang=de&topic=ech&layers=WMS%7C%7CHalbtonvariante%7C%7Chttp:%2F%2Fwww.webatlasde.de%2Farcgis%2Fservices%2FMaps4BW%2FMapServer%2FWMSServer%7C%7CHalbtonvariante%7C%7C1.3.0%7C%7Ctrue&E=2646926.31&N=1285318.00&zoom=3)
Fix #3830 [Display a WMTS on 3d if a matrixset WGS84 or 3857 is available](https://mf-geoadmin3.int.bgdi.ch/teo_wmts3d/index.html?lang=de&topic=ech&bgLayer=ch.swisstopo.pixelkarte-grau&layers=WMTS%7C%7Ccore003_feathering_mixed%7C%7Chttp:%2F%2Fcidportal.jrc.ec.europa.eu%2Fcopernicus%2Fservices%2Ftile%2Fwmts%2F1.0.0%2FWMTSCapabilities.xml,ch.swisstopo.swissnames3d&lon=8.23818&lat=42.88070&elevation=435327&heading=360.000&pitch=-43.894)

### CODE REVIEW

**Comment by loicgasser** [comment]
Be careful, there are several important regressions in 3.5, notably with ImageVector, which will be deprecated in favour of a parameter `renderMode: 'image'.` in ol.layer.Vector.

**Comment by oterral** [comment]
Where do you see that ?

**Comment by oterral** [comment]
Ah yes just saw the PR

**Comment by gjn** [comment]
Great! Why not havin the coverage report page in production? :=)

The general testlink and the asit VD testlink does show translations, the last 2 testlinks do not. Why?

File: Makefile [hunk 1]
```diff
@@ -348,6 +349,7 @@ olcesium: .build-artefacts/ol-cesium
 	git show; \
 	git submodule update --recursive --init --force; \
 	cd ol; \
+	git remote | grep oterral || git remote add oterral git://github.com/oterral/ol3; \
```
Comment (gjn) [inline Makefile]: Reminder not to forget this once ol3 is merged.

File: src/locales/it.json [hunk 2]
```diff
@@ -229,6 +229,7 @@
 "kml": "KML",
 "lang_chooser_label": "Lingua",
 "last_available_year": "Last year",
+"layer_cant_be_displayed_in_3d": "",
```
Comment (gjn) [inline src/locales/it.json]: Could translators directly work here?

**Comment by oterral** [comment]
> Great! Why not havin the coverage report page in production? :=)
Because to access it I use an apache alias which we only have in dev.

> The general testlink and the asit VD testlink does show translations, the last 2 testlinks do not. Why?
Didin't notice that. I will try to find why.

File: src/locales/it.json (hunk 2 above)
Comment (oterral) [reply src/locales/it.json]: I guess yes

**Comment by oterral** [comment]
@gjn can I merge?

**Pull request merged**